src/py/flwr/superexec/executor.py

```diff
@@ -0,0 +1,59 @@
+# Copyright 2024 Flower Labs GmbH. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Execute and monitor a Flower run."""
+
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from subprocess import Popen
+from typing import Optional
+
+
+@dataclass
+class RunTracker:
+    """Represents a Flower run (composed of a run_id and the associated process)."""
+
+    run_id: int
+    proc: Popen  # type: ignore
+
+
+class Executor(ABC):
+    """Execute and monitor a Flower run."""
+
+    @abstractmethod
+    def start_run(
+        self,
+        fab_file: bytes,
+        ttl: Optional[float] = None,
```

Comment:
@danieljanes We don't use it currently, I just left it there in case we want to do something with it in the future (like not executing stale messages). Should I remove it?

Reply:
@charlesbvll let's remove it for now